cmake/CheckHeaders.cmake

```diff
@@ -0,0 +1,113 @@
+# a poor's alternative of iwyu
+#
+# Header files should be self-contained. In general, the source files should
+# not have to adhere to special conditions to include them. For instance,
+# they don't need to include other header files for using a header file, or
+# to define certain macro(s) for using it. But the macros are allowed to be
+# used to its behavior though.
+#
+# include-what-you-use is able to check if all symbols used by a certain header
+# are defined by the headers *directly* included by that header by parsing the
+# header source file using libclang. But we don't always have iwyu installed.
```

Comment:
> I don't understand all these references to "iwyu".

IMHO, it's quite popular. so i assume that developer reading this would question, why do you use this homebrew solution? and how it is different from iwyu.

> If this tool is so fantastic, we can use it.

it relies on clang-dev. and it is **much** more picky than this poor man's solution. at this moment, i don't think we should use it for a sanity check

> If it's just a trivial shell script on top of the regular compiler (which it should be!!) then we don't need to mention it at all...

no, it is not.

Reply:
I never personally heard about this specific tool. The way you describe it it sounds both wrong (uses clang when we need to also fully support gcc) and annoying (more picky means more annoying!), so we shouldn't even strive to be compatible with it. I personally wouldn't even mention it. But I guess you can, if you want, we don't pay github by lines of comments stored ;-)

Reply:
i see. i wanted to be prepared for the questions. but it seems it's just me who is paranoid. i am happy to remove this bulk of text.

Reply:
forgot to mention, clang-dev is a build-time dependency of iwyu. as it is not active maintained even in fc37, i wanted to note this dependency to provide more background on "we can use it".